nbextensions/appCell2/widgets/appCellWidget-fsm.js

```diff
@@ -950,45 +738,29 @@ define([], () => {
                     ],
                 },
             },
-            next: [
-                {
-                    mode: 'error',
-                    stage: 'runtime',
-                },
-                {
-                    mode: 'editing',
-                    params: 'complete',
-                    code: 'built',
-                },
-                {
-                    mode: 'internal-error',
-                },
-            ],
+            next: [STATE.RUNTIME_ERROR, STATE.EDITING_COMPLETE, STATE.INTERNAL_ERROR],
         },
         // A fatal error represents an app cell which cannot operate.
         {
-            state: {
-                mode: 'internal-error',
-            },
+            state: STATE.INTERNAL_ERROR,
             ui: {
                 tabs: uiTabs.error,
                 actionButton: {
                     name: 'resetApp',
                 },
             },
             next: [
-                {
-                    mode: 'internal-error',
-                },
+                STATE.INTERNAL_ERROR,
                 // We will let a user attempt to reset the app.
-                {
-                    mode: 'new',
-                },
+                STATE.NEW,
+                STATE.EDITING_INCOMPLETE,
+                STATE.EDITING_COMPLETE,
```

Comment:
the app goes to state `editing_incomplete` or `editing_complete` on reset -- not `new`